src/ui/src/components/blueprints/abstract/BlueprintsNode.vue

```diff
@@ -145,21 +145,44 @@ const completionStyle = computed(() => {
 	);
 });
 
-const latestKnownOutcome = computed(() => {
+const latestRun = computed(() => {
 	const logEntries = wfbm.getLogEntries();
+	const runId = logEntries.find((entry) => {
+		return !!entry.blueprintExecution;
+	})?.blueprintExecution?.runId ?? null;
+	return logEntries.filter((entry) => {
+		return entry?.blueprintExecution?.runId === runId;
+	});
+});
 
-	for (let i = 0; i < logEntries.length; i++) {
-		const logEntry = logEntries[i];
-		const we = logEntry.blueprintExecution;
-		if (!we) continue;
-		for (let j = 0; j < we.summary.length; j++) {
-			const item = we.summary[j];
-			if (item.componentId !== component.value.id) continue;
-			return item.outcome;
-		}
-	}
+const outcomeSeverity = {
+	in_progress: 5,
+	error: 4,
+	success: 3,
+	cancelled: 2,
+	skipped: 1,
+	none: 0,
+}
 
-	return null;
+const latestKnownOutcome = computed(() => {
+	const executionLogs = latestRun.value.map((entry) => {
+		return entry.blueprintExecution;
+	}).filter(Boolean);
+	let outcome = 'none';
+	executionLogs.forEach((log) => {
+		log.summary.forEach((item) => {
+			if (item.outcome && item.componentId === component.value.id) {
+				const severity = (!Object.keys(outcomeSeverity).includes(item.outcome) ? outcomeSeverity.success : outcomeSeverity[item.outcome]);
```

Comment:
It's more readable to get the truthy value first
```suggestion
				const severity = Object.keys(outcomeSeverity).includes(item.outcome) ? outcomeSeverity[item.outcome] : outcomeSeverity.success;
```